components/collator/benches/bench.rs

```diff
@@ -0,0 +1,89 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::time::Duration;
+
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+
+use icu::collator::*;
+use icu::locid::Locale;
+
+pub fn collator_with_locale(criterion: &mut Criterion) {
+    // Load file content in reverse order vector.
+    let content_latin: (&str, Vec<&str>) = (
+        "TestNames_Latin",
+        include_str!("data/TestNames_Latin.txt")
+            .split('\n')
+            .rev()
+            .collect::<Vec<&str>>(),
+    );
+    let content_asian: (&str, Vec<&str>) = (
+        "TestNames_Asian",
+        include_str!("data/TestNames_Asian.txt")
+            .split('\n')
+            .rev()
+            .collect(),
+    );
+    let content_russian: (&str, Vec<&str>) = (
+        "TestNames_Russian",
+        include_str!("data/TestNames_Russian.txt")
+            .split('\n')
+            .rev()
+            .collect(),
+    );
+
+    let performance_parameters = [
+        ("en_US", vec![&content_latin]),
+        ("fr_FR", vec![&content_latin]),
```

Comment:
`fr-FR` and `en-US` both map to the root collation. Is it intentional to bench the same thing twice to see if the two give the same results, or should the second collation be a collation for some other Latin-script language that does have a tailoring? (The data is almost exclusively ASCII, so benching a collation with tailoring would pretty much bench the fallback to root and contractions not ending up contracting.)

Since `zh-CN` is tested with Latin-only input, I guess in practice it acts as a collation with tailoring with Latin contractions (for Pinyin diacritics) with most things falling back to root here.

Reply:
So what I did was was picking the 3 files Latin, Asian, Russian and picked some already mapped existing bench.

https://github.com/unicode-org/icu/blob/main/icu4j/perf-tests/collationperf.pl

If you think I think I should pull more/rearrange, please advise :)

I'll remove the fr-FR, and no we should not bench the same things. 
I've initially left it because french language have accents. (but could have pull any other language really.)

Reply:
en-US, fr-FR, and de-DE are all the same thing. ru-RU is also the same collation but with script reordering post-processing, which shouldn't dominate a benchmark. However, it makes sense to test with non-Latin (in this case Russian) input.

Is suggest testing:
* und with Latin data (the same as en-US, fr-FR, and de-DE, but und makes it clear the root is being benched)
* da-DK with Latin data (de-u-co-phonebk is unlikely to be much different perf-wise)
* fr-CA with Latin data (the ICU test is so old that its fr-FR probably meant to test backward second level, which remains only in fr-CA these days)
* ja with Japanese data
* zh-u-co-pinyin (the same as zh-CN) with Chinese data
* zh-u-co-stroke (the same as zh-TW) with Chinese data
* ru with Russian data
* th with Thai data
* ko with Korean data

Reply:
I think we have all the needed data